src/store.rs

```diff
@@ -471,335 +587,287 @@ impl<'a> Run<'a> {
+    /// The publication point is identified through the rsync URI of its
+    /// manifest.
     ///
-    /// If the update succeeds, the method returns whatever it is the closure
-    /// returned. If it fails, it was either aborted by the closure or there
-    /// was a database error. You can check whether it was indeed aborted by
-    /// calling [`was_aborted`](UpdateError::was_aborted) on the error.
-    pub fn update_point<T, F> (
-        &self, manifest: &uri::Rsync, op: F
-    ) -> Result<T, UpdateError>
-    where
-        F: Fn(&mut RepositoryUpdate) -> Result<T, UpdateError>
-    {
-        let mut update = RepositoryUpdate::new(manifest);
-        let res = op(&mut update)?;
-        (&self.manifest_tree, &self.object_tree).transaction(|(mt, ot)| {
-            if let Some(ref new_manifest) = update.new_manifest {
-                mt.insert(
-                    manifest.as_str(), new_manifest
-                )?;
-            }
-            ot.apply_batch(&update.objects).map_err(Into::into)
-        })?;
-        Ok(res)
+    /// A stored point instance will be returned whether there actually is
+    /// information stored for the point or not. You can use
+    /// [`StoredPoint::exists`] to check if there is previously stored
+    /// information.
+    pub fn get_point(
+        &self, manifest_uri: &uri::Rsync
+    ) -> Result<StoredPoint<'a>, Failed> {
+        StoredPoint::open(
+            self.store, self.point_path(manifest_uri), self.is_rrdp
+        )
     }
 
     /// Completely removes a publication point.
     ///
-    /// The publication point to be updated is given via its manifest’s
+    /// The publication point to be removed is given via its manifest’s
     /// signedObject URI.
-    ///
-    /// If the publication point exists, its manifest and all its objects are
-    /// removed from the database.
     pub fn remove_point(
-        &self, manifest: &uri::Rsync
-    ) -> Result<(), Failed> {
-        let mut batch = sled::Batch::default();
-        for key in self.object_tree.scan_prefix(
-            Self::object_key(manifest, "")
-        ).keys() {
-            batch.remove(key?);
-        }
-        (&self.manifest_tree, &self.object_tree).transaction(|(mt, ot)| {
-            mt.remove(manifest.as_str())?;
-            ot.apply_batch(&batch)?;
-            Ok(())
-        }).map_err(|err| {
-            match err {
-                TransactionError::Abort(()) => {
-                    unreachable!() // Or is it?
-                }
-                TransactionError::Storage(err) => {
-                    error!("Failed to update storage: {}", err);
-                    Failed
-                }
-            }
-        })
+        &self, manifest_uri: &uri::Rsync
+    ) -> Result<(), io::Error> {
+        fs::remove_file(self.point_path(manifest_uri))
+    }
+
+    fn point_path(&self, manifest_uri: &uri::Rsync) -> PathBuf {
+        self.path.join(
+            format!(
+                "rsync/{}/{}/{}",
```

Comment:
Still not sure what the intended directory structure is, whether objects fetched via rrdp should actually be stored in ./rsync. Also see related comments on collector/rrdp.rs

Reply:
Clear now, from the other comments.